src/Http/Routing/src/DependencyInjection/RoutingServiceCollectionExtensions.cs

```diff
@@ -131,8 +138,8 @@ public static IServiceCollection AddRoutingCore(this IServiceCollection services
         ArgumentNullException.ThrowIfNull(services);
         ArgumentNullException.ThrowIfNull(configureOptions);
 
-        services.AddRouting();
         services.Configure(configureOptions);
+        services.AddRouting();
```

Comment:
Why did the order change?

Reply:
See discussions in the PR, but @eerhardt was pointing out that I had changed the order. I said I don't think it matters but swapped it back. Arguably doing configure last is best because if there is something where we don't detect the user previously overriding a default then we could screw up.

In this case I think it is safe because the only thing we do in ```AddRouting(this IServicesCollection services)``` is add the constraint after first checking it wasn't overridden.